PureBasicIDE/Preferences.pb

```diff
@@ -2968,11 +2967,13 @@ Procedure OpenPreferencesWindow()
   Next i
   
   Restore DefaultColorSchemes
-  Read.l NbSchemes
+  NbSchemes = 0   ; Number of Default Color Schemes is automatically counted below
   
   CurrentScheme = -1
-  For i = 1 To NbSchemes
-    Read.s Name$
+  i = 0
+  Read.s Name$
+  While Name$ <> "" ; Empty Name$ indicates end of color schemes
+    NbSchemes + 1
```

Comment:
`i` was used in the `For` loop because a counter is needed to stop the loop and to set `CurrentScheme`.
With the new behaviour:
- The loop exit no longer use a counter
- The new var `NbSchemes`  also act as a counter but is not used in the loop.
That why I think you should remove `i` and replace it by `NbSchemes` to set `CurrentScheme`. (The `NbSchemes + 1` statement can be moved after the `If IsMatch ... EndIf` statement to keep the `CurrentScheme` assignment simple without addition nor subtraction.

Reply:
True, we don't need two variables incrementing in the same `While` loop. I'll simplify it and push branch again.

Side note: I confirmed **this PR is a bugfix for the SpiderBasic build**. When the "Dark Mode" theme was added, the count in the SpiderBasic section was not updated. So if you build and run the SB IDE, the bottom scheme will be Dark Mode's colors but labeled "Accessibility". This PR fixes that.